src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,104 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int GROW_COEFFICIENT = 2;
+    private Node<K, V>[] elements;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        elements = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (elements.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> existingNode = elements[index];
+        Node<K, V> newNode = new Node<>(getHash(key), key, value, null);
+        if (Objects.isNull(existingNode)) {
+            elements[index] = newNode;
+        }
+        while (Objects.nonNull(existingNode)) {
+            if (Objects.equals(existingNode.key, key)) {
+                existingNode.value = value;
+                return;
+            }
+            if (Objects.isNull(existingNode.next)) {
+                existingNode.next = newNode;
+                break;
+            }
+            existingNode = existingNode.next;
+        }
+        if (++size > threshold) {
+            resize();
```

Comment:
It's better to check if the map needs resizing at the beginning of the `put` method. This avoids unnecessary work in case the map needs resizing.